src/storage/src/client.rs

```diff
@@ -358,11 +367,16 @@ impl InsertObject {
         );
 
         let builder = self.inner.apply_auth_headers(builder).await?;
-        let content = match self.request.data {
-            Some(Data::ChecksummedData(data)) => data.content,
-            _ => unreachable!("content for the checksummed data is set in the constructor"),
-        };
-        let builder = builder.body(content);
+
+        let stream = Box::pin(unfold(Some(self.payload), move |state| async move {
+            if let Some(mut payload) = state {
+                if let Some(n) = payload.next().await {
+                    return Some((n, Some(payload)));
```

Comment:
comment: This took me a second (probably due to forgetting the `unfold` interface). `n` is `Result<bytes::Bytes>`. I think you have `n` as a shortened `next`, but `n` made me think of "number of bytes".

Consider spelling out `next` or calling it `bytes`.